packages/inngest/src/components/InngestCommHandler.ts

```diff
@@ -722,12 +722,35 @@ export class InngestCommHandler<
           (await getQuerystring("processing run request", queryKeys.StepId)) ||
           null;
 
+        const headersToForward = [
+          headerKeys.TraceParent,
+          headerKeys.TraceState,
+        ];
+
+        const headers = await headersToForward.reduce<
+          Promise<Record<string, string>>
+        >(async (acc, header) => {
+          const value = await actions.headers(
+            `fetching ${header} for forwarding`,
+            header
+          );
+          if (value) {
+            return {
+              ...(await acc),
+              [header]: value,
+            };
+          }
+
+          return acc;
+        }, Promise.resolve({}));
```

Comment:
I can't read JS. Does this means take the headers passed in, if there's nothing, default to an empty map?

Reply:
@darwin67 Correct.

It looks a bit more complex as this hits the user's serve handler to access each header, where the function for retrieving that header may be async. This pattern is for making those calls in parallel and then rolling up the results into a map at the end, though it's an abstract one.

I'll refactor to make it more explicit.